lib/matplotlib/_docstring.py

```diff
@@ -3,6 +3,31 @@
 from . import _api
 
 
+def kwarg_doc(text):
+    """
+    Decorator for defining the documentation used when artist properties are
+    passed through as keyword arguments.
+
+    See e.g. the `**kwargs` section in `.Axes.text`.
+
+    The given text is stored in a privat variable ``_kwarg_doc`` on the method.
+    It is used for generating the kwdoc list for artists, which we
+    auto-generate through docstring interpolation, e.g. via
+    ``%(Line2D:kwdoc)s``.
+
+    The text should contain the supported types as well as the default value
+    if applicable, e.g.:
+
+        @_docstring.kwarg_doc("bool, default: :rc:`text.usetex`")
+        def set_usetex(self, usetex):
```

Comment:
I agree it's pretty mysterious from this text.  Suggest more explanation, though the following is probably incorrect.  The docstring should motivate why this is helpful and what effect it has on the documentation.  

```suggestion
    """
    Decorator on property setters for defining the documentation used when artist 
    properties are passed through as keyword arguments.

    We generate lists of kwargs by inspecting the properties on a class, and then refer to them 
    in the documentation as, for example, ``%(Text:kwdoc)s``, which is used in the `**kwargs` 
    section in `.Axes.text`.  However, sometimes the setter function parameter description is
    not what we want in this generated list.  

    This method adds the given text to a private variable ``_kwarg_doc`` on the class.
    It will be used for generating the kwdoc list for artists instead of the parameter docs.  

    The text should contain the supported types as well as the default value
    if applicable, e.g.:

        @_docstring.kwarg_doc("bool, default: :rc:`text.usetex`")
        def set_usetex(self, usetex):
```